src/server/project.ts

```diff
@@ -1070,7 +1070,7 @@ namespace ts.server {
                 this.lastCachedUnresolvedImportsList = undefined;
             }
 
-            const isFirstLoad = this.projectProgramVersion === 0;
+            const isFirstLoad = this.projectProgramVersion === 0 && hasNewProgram;

```

Comment:
This appears to do the right thing, but it seems like the name and the meaning no longer agree (presumably, it's the first load whether or not there's a program).

Reply:
Fair enough; I’ll make the name more specific.